lib/rspec/core/formatters/exception_presenter.rb

```diff
@@ -124,16 +140,24 @@ def exception_class_name(exception=@exception)
         end
 
         def failure_lines
-          @failure_lines ||=
-            begin
-              lines = []
-              lines << failure_slash_error_line unless (description == failure_slash_error_line)
-              lines << "#{exception_class_name}:" unless exception_class_name =~ /RSpec/
-              encoded_string(exception.message.to_s).split("\n").each do |line|
-                lines << "  #{line}"
-              end
-              lines
-            end
+          @failure_lines ||= failure_slash_error_lines + exception_lines
+        end
+
+        def failure_slash_error_lines
+          read_failed_lines.tap do |lines|
+            least_indentation = lines.map { |line| line.match(/^[ \t]*/)[0] }.min
+            lines.map! { |line| '  ' + line.gsub(/^#{least_indentation}/, '') }
```

Comment:
I think this can use `sub` instead of `gsub`, which might be faster (as gsub may try to find more matches in the string....) and is more explicit about it doing only one substitution.


Also, would this line be equivalent to this?

``` ruby
lines.map! { |line| line.gsub(/^#{least_indentation}/, '  ') }
```

It seems odd to strip all of the common indentation, then add back two spaces of indentation, rather than just replacing the common indentation with two spaces.

Or is it necessary to do it in two steps for some reason I'm not seeing?